tasks/message/gasoutputs.go

```diff
@@ -125,6 +132,35 @@ func (p *GasOutputsProcessor) processItem(ctx context.Context, node lens.API, it
 	stop := metrics.Timer(ctx, metrics.ProcessingDuration)
 	defer stop()
 
+	// Note: this item will only be processed if there are receipts for
+	// it, which means there should be a tipset at height+1.  This is only
+	// used to get the destination actor code, so we don't care about side
+	// chains.
+	child, err := node.ChainGetTipSetByHeight(ctx, abi.ChainEpoch(item.Height+1), types.NewTipSetKey())
+	if err != nil {
+		return xerrors.Errorf("Failed to load child tipset: %w", err)
+	}
+
+	st, err := state.LoadStateTree(node.Store(), child.ParentState())
+	if err != nil {
+		return xerrors.Errorf("load state tree when gas outputs for %s: %w", item.Cid, err)
+	}
+
+	dstAddr, err := address.NewFromString(item.To)
+	if err != nil {
+		return xerrors.Errorf("parse to address failed for gas outputs in %s: %w", item.Cid, err)
+	}
+
+	var dstActorCode cid.Cid
+	dstActor, err := st.GetActor(dstAddr)
+	if err != nil {
+		if !errors.Is(err, types.ErrActorNotFound) {
+			return xerrors.Errorf("get destination actor for gas outputs %s failed: %w", item.Cid, err)
+		}
+	} else {
+		dstActorCode = dstActor.Code
+	}
+
```

Comment:
I wish there was a better way, but this should be ok if the state is cached at some layer.

Reply:
https://github.com/filecoin-project/sentinel-visor/pull/249 is much more efficient but will be a few days before it's ready. Currently checking that the data it produces matches the existing data.

Reply:
yeah, still need to support it in the "standard" way which I trust will become way faster with store improvement and block caching at some point.